common/models/acl.js

```diff
@@ -457,10 +486,14 @@ module.exports = function(ACL) {
 
         var resolved = self.resolvePermission(effectiveACLs, req);
         if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
+          resolved.permission = defaultPermission || ACL.ALLOW;
         }
         debug('---Resolved---');
         resolved.debug();
+        // add authorizedRoles to remotingContext if overall resolved permission status is ALLOW
+        if (resolved.permission === ACL.ALLOW && remotingContext) {
+          remotingContext.args.options.authorizedRoles = authorizedRoles;
+        };
         return callback(null, resolved);
```

Comment:
only add the authorizedRoles array to remotingContext if the overall resolved permission is ALLOW
the condition should not be necessary considering the request would return a 401 if resolved permission is DENY, but it's probably safer if we consider custom middlewares running just after the auth stage (would they run if the auth result is 401?)